include/alpaka/core/Assert.hpp

```diff
@@ -81,7 +88,7 @@ namespace alpaka
 #ifdef NDEBUG
                     alpaka::ignore_unused(lhs);
 #else
-                    ALPAKA_ASSERT(TLhs::value > lhs);
+                    ALPAKA_ASSERT_OFFLOAD(TLhs::value > lhs);
```

Comment:
@psychocoderHPC for example, here the `alpaka::ignore_unused(lhs)` would need to become unconditional for `ALPAKA_ASSERT_OFFLOAD` to work without the `alpaka::ignore_unsused` fallback definition (or check `ALPAKA_OFFLOAD_ASSUME_HOST`, which would defeat the purpose of the macro).